File: test-serenity2.Web/Modules/Games/Platforms/PlatformsForm.cs
using Serenity.ComponentModel;

namespace test_serenity2.Games.Forms;

[FormScript("Games.Platforms")]
[BasedOnRow(typeof(PlatformsRow), CheckNames = true)]
public class PlatformsForm
{
    public string Name { get; set; }
    public int Year { get; set; }
}

File: test-serenity2.Web/Modules/ServerTypes/Games/PlatformsForm.ts
import { StringEditor, IntegerEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface PlatformsForm {
    Name: StringEditor;
    Year: IntegerEditor;
}

export class PlatformsForm extends PrefixedContext {
    static readonly formKey = 'Games.Platforms';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PlatformsForm.init)  {
            PlatformsForm.init = true;

            var w0 = StringEditor;
            var w1 = IntegerEditor;

            initFormType(PlatformsForm, [
                'Name', w0,
                'Year', w1
            ]);
        }
    }
}